record-and-playback/core/lib/recordandplayback.rb

```diff
@@ -150,6 +150,32 @@ def self.monotonic_clock()
     return (AbsoluteTime.now * 1000).to_i
   end
 
+  def self.download(url, output)
+    BigBlueButton.logger.info "Downloading #{url} to #{output}"
+
+    uri = URI.parse(url)
+    if ["http", "https", "ftp"].include? uri.scheme
+      command = "wget -q --spider #{url}"
```

Comment:
Do we really need to run wget here (or curl below)? Ruby can do http requests natively.

Reply:
@kepstin it's for simplicity, since wget and curl are already installed in a BigBlueButton server.

Reply:
I really would prefer that you use ruby modules to do http request rather than shelling out to external processes. You could also avoid doing two separate requests then - the `wget --spider` is unnecessary since you can handle a received 404 directly.

If you really want to avoid using ruby's http support, then you *must* fix these commands so that the arguments are safetly passed to the command - using a single string is unsafe (a URL containing an unescaped space could do bad things.)

Reply:
@kepstin Do you mean using ruby Net module? This one http://ruby-doc.org/stdlib-2.6.2/libdoc/net/http/rdoc/Net/HTTP.html ?

Reply:
Yes, the ruby Net::HTTP (`net/http`) module would work fine for this. Note that on Ubuntu 16.04 we only have ruby 2.3, so use the correct reference docs: https://ruby-doc.org/stdlib-2.3.0/libdoc/net/http/rdoc/Net/HTTP.html - there may have been additional features added in newer versions.